pan3d/explorers/slice_explorer.py

```diff
@@ -0,0 +1,441 @@
+        origin = [
+            float(extents[0][0] + (extents[0][1] - extents[0][0]) / 2), 
+            float(extents[1][0] + (extents[1][1] - extents[1][0]) / 2), 
+            float(extents[2][0] + (extents[2][1] - extents[2][0]) / 2)
+        ]
+
+        camera = self.renderer.GetActiveCamera()
+        pos = camera.GetPosition()
+        norm = np.linalg.norm(np.array(origin) - np.array(pos))
+        print(f"Origin and norm : {origin, norm}")
+        if self.state.view_mode == '3D':
+            camera.SetPosition(norm, norm, norm)
+            camera.SetViewUp(0., 1., 0.)
+        elif self.state.view_mode == '2D':
+            self.renderer.RemoveActor(self.outline_actor)
+            self.renderer.RemoveActor(self.data_actor)
+            if slice_i == 0:
+                camera.SetPosition(norm, origin[1], origin[2])
+                camera.SetViewUp(0., 1., 0.)
+            elif slice_i == 1:
+                camera.SetPosition(origin[0], norm, origin[2])
+                camera.SetViewUp(0., 0., 1.)
+            elif slice_i == 2:
+                camera.SetPosition(origin[0], origin[1], norm)
+                camera.SetViewUp(0., 1., 0.)
+            camera.SetFocalPoint(origin[0], origin[1], origin[2])
+            camera.OrthogonalizeViewUp()
+        self.renderer.SetActiveCamera(camera)
+        self.renderer.ResetCamera()
+        self.ctrl.view_update()
+
+    @change("data_var", "time_active", "dimval")
+    def on_data_change(self, **_):
+        dimval    = self.state.dimval
+        slice_dim = self.state.slice_dim
+        normal      = [0, 0, 0]
+        slice_i     = 0 if self.dims[slice_dim] == 'x' else 1 if self.dims[slice_dim] == 'y' else 2
+        normal[slice_i] = 1
+        
+        extents = list(self.extents.values())
+        origin = [
+            float(extents[0][0] + (extents[0][1] - extents[0][0]) / 2), 
+            float(extents[1][0] + (extents[1][1] - extents[1][0]) / 2), 
+            float(extents[2][0] + (extents[2][1] - extents[2][0]) / 2)
+        ] 
+        origin[slice_i] = dimval
+        
+        self.plane.SetOrigin(origin)
+        self.plane.SetNormal(normal)
+
+        self.cutter.SetInputData(self.t_slice)
+        self.cutter.SetCutFunction(self.plane)
+        self.cutter.Update()
+        output = self.cutter.GetOutput()
+        vrange = output.GetPointData().GetArray(self.state.data_var).GetRange()
+        self.slice_mapper.SetScalarRange(float(vrange[0]), float(vrange[1]))
+        self.state.varmin = float(vrange[0])
+        self.state.varmax = float(vrange[1])
+        self.sbar_actor.SetLookupTable(self.slice_mapper.GetLookupTable())
+
+        if self.state.view_mode == "2D":
+            self.on_view_mode_change("2D")
+            self.renderer.ResetCamera()
+
+        self.ctrl.view_update()
+
+    @change("outline", "tdata")
+    def on_rep_change(self, outline, tdata, **_):
+        if outline: 
+            self.renderer.AddActor(self.outline_actor)
```

Comment:
This is a good point, I think adding or removing actors would be helpful in large datasets as we are allowing users to have the surface rendering of the original data with some value of opacity set. Adding/removing will help control the memory of rendering artifacts, thus, also helping with performance? I think this will need a performance study as I cannot make these gusses.

Reply:
I'm not talking about opacity. I don't agree with your performance assumption. Technically the visibility does the same thing as add/remove but without mutating a vtkCollection.